### PR TITLE
Refactor/2318/migrate attribute type selector

### PR DESCRIPTION
**Pull request by shaman-apprentice**
- Removed type `"edges"` as it was unused.
- Marginal changes in look like menu items `font-size: 15px` -> `font-size: 16px` as it looks better to me. I think not worth any discussion or mention in changelog.

### CODE REVIEW

**Review by BridgeAR — COMMENTED** [review]
The code LGTM. There is just a tiny regression in visualizing the buttons (they are not centered properly anymore):

Now:
image
Before:
image

**Comment by shaman-apprentice** [comment]
> The code LGTM. There is just a tiny regression in visualizing the buttons (they are not centered properly anymore):
> 
> Now: image Before: image

Thx for the catch! Fixed with f9f2419

File: visualization/app/material/matMenu.scss [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+		line-height: 20px;
+	}
+
+	// overwrite global app.scss style
+	button {
+		margin: unset;
+	}
+}
+
+.mat-menu-item {
```
Comment (Christian-Eberhard) [inline visualization/app/material/matMenu.scss:14]: .mat-menu-item is a child of .mat-menu-content and could be in the same paranthesis level as .menu-header above

File: visualization/app/material/matMenu.scss [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	font-size: 16px;
```
Comment (Christian-Eberhard) [inline visualization/app/material/matMenu.scss:15]: Generally id preffer pt over px, when working with fonts

**Review by Christian-Eberhard — CHANGES_REQUESTED** [review]

File: visualization/app/codeCharta/ui/attributeTypeSelector/attributeTypeSelector.component.scss [hunk 3]
```diff
@@ -1,33 +1,22 @@
...
 		width: 24px;
-		vertical-align: top;
-
-		.md-fab {
-			min-height: 0;
-			width: 100%;
-			height: 100%;
-			margin: 0;
-			text-transform: unset !important;
+		min-width: 0 !important;
```
[the remark this answers is not in the document]
Comment (shaman-apprentice) [reply visualization/app/codeCharta/ui/attributeTypeSelector/attributeTypeSelector.component.scss:9]: I agree that `!important` is evil :D But if needed I wouldn't invest too much time into it, as the main goal of the Angular migration is not to improve the css. Although the complete property has become unnecessary through some refactoring. So I delete it with next push

File: visualization/app/codeCharta/ui/attributeTypeSelector/attributeTypeSelector.component.scss [hunk 4]
```diff
@@ -1,33 +1,22 @@
...
+		padding: 0;
```
[the remark this answers is not in the document]
Comment (shaman-apprentice) [reply visualization/app/codeCharta/ui/attributeTypeSelector/attributeTypeSelector.component.scss:11]: we also need to readjust top of .symbol due to vertical-align. Therefore not using the suggestion

File: visualization/app/codeCharta/ui/attributeTypeSelector/attributeTypeSelector.component.scss [hunk 5]
```diff
@@ -1,33 +1,22 @@
...
 
-			.symbol {
-				position: absolute;
-				top: -16px;
-				left: 8px;
-			}
+		&:hover {
+			background-color: rgb(200, 200, 200);
```
[the remark this answers is not in the document]
Comment (shaman-apprentice) [reply visualization/app/codeCharta/ui/attributeTypeSelector/attributeTypeSelector.component.scss:13]: done

File: visualization/app/codeCharta/ui/attributeTypeSelector/attributeTypeSelector.component.html [hunk 6]
```diff
@@ -1,21 +1,12 @@
...
-	</md-menu-content>
-</md-menu>
+<div class="small-action-button transparent">
+	<button mat-button [matMenuTriggerFor]="menu" title="Select attribute type" class="menu-button">
+		<span class="symbol">{{ metric | aggregationSymbolPipe: (getAttributeTypeOfNodesByMetric$ | async) }}</span>
+	</button>
+</div>
+
+<mat-menu #menu>
+	<div class="menu-header" mat-menu-item disabled>Set global aggregation for rloc</div>
```
[the remark this answers is not in the document]
Comment (shaman-apprentice) [reply visualization/app/codeCharta/ui/attributeTypeSelector/attributeTypeSelector.component.html:8]: The advantage of this standard isn't clear to me. Anyway, lets not spend time in something which we might do, while "just" migrating to Angular, which is already a very very huge task :P

File: visualization/app/material/matMenu.scss (hunk 1 above)
Comment (shaman-apprentice) [reply visualization/app/material/matMenu.scss:14]: done

File: visualization/app/material/matMenu.scss (hunk 2 above)
Comment (shaman-apprentice) [reply visualization/app/material/matMenu.scss:15]: Generally I agree. But as we mix it up everywhere and have no concept for it so far, I don't want to invest any time into it while migrating.

File: visualization/app/codeCharta/ui/attributeTypeSelector/attributeTypeSelector.component.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (Christian-Eberhard) [reply visualization/app/codeCharta/ui/attributeTypeSelector/attributeTypeSelector.component.html:8]: Text just in a div has no semantic meaning for text, its used for layout, while span is used to style texts. If that makes more sense

**Review by Christian-Eberhard — APPROVED** [review]

**Pull request merged**